src/VisualStudio/Core/Test.Next/Services/VisualStudioDiagnosticAnalyzerExecutorTests.cs

```diff
@@ -120,6 +122,33 @@ End Sub
             }
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.RemoteHost)]
+        public async Task TestCancellationOnSessionWithSolution()
+        {
+            var code = @"class Test { void Method() { } }";
+
+            using (var workspace = CreateWorkspace(LanguageNames.CSharp, code))
+            {
+                var solution = workspace.CurrentSolution;
+                var solutionChecksum = await solution.State.GetChecksumAsync(CancellationToken.None);
+
+                var source = new CancellationTokenSource();
+                var connection = new MyConnection(source);
+                try
+                {
+                    await SessionWithSolution.CreateAsync(connection, solution, source.Token);
```

Comment:
:grey_exclamation: Remove the try/catch here. Use Assert.ThrowsAsync if the code throws.

Reply:
sure.